src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -32,6 +32,8 @@ public class TagHelperDescriptorFactory
                 RegexOptions.None,
                 Constants.RegexMatchTimeout);
 
+        private static readonly char[] UnsupportedAttributeSelectors = new[] { '~', '|', '*' };
```

Comment:
Oh and also, keep in mind, everything prior to the `=` not captured by the blacklist still runs through the normal attribute/tag name validation system which notifies invalid usage of several other characters. https://github.com/aspnet/Razor/blob/a0164b005d9bdd0ee59de3c90c38499fdc8ebd15/src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs#L369-L399